src/cascadia/TerminalSettingsEditor/GlobalAppearance.cpp

```diff
@@ -144,30 +161,41 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
 
     winrt::Windows::Foundation::IInspectable GlobalAppearance::CurrentLanguage()
     {
-        if (_currentLanguage.empty())
+        if (_currentLanguage)
         {
-            _currentLanguage = winrt::Windows::Globalization::ApplicationLanguages::PrimaryLanguageOverride();
-            if (_currentLanguage.empty())
-            {
-                _currentLanguage = systemLanguageTag;
-            }
+            return _currentLanguage;
+        }
+
+        if (!LanguageSelectorAvailable())
+        {
+            _currentLanguage = {};
+            return _currentLanguage;
+        }
+
+        // NOTE: PrimaryLanguageOverride throws if this instance is unpackaged.
+        auto currentLanguage = winrt::Windows::Globalization::ApplicationLanguages::PrimaryLanguageOverride();

```

Comment:
```suggestion
        const auto currentLanguage = winrt::Windows::Globalization::ApplicationLanguages::PrimaryLanguageOverride();
```

Reply:
This variable cannot be const as I conditionally overwrite its value in the if condition right after.

Reply:
BTW gcc allows you to omit the second operand to a ternary operator and write:
```gcc
const auto currentLanguage = PrimaryLanguageOverride() ?: winrt::hstring{...};
```

MSVC doesn't support this (very useful! ...but non-standard) extension unfortunately.

Reply:
i wish for that all the time :smile: